osu.Game/Screens/SelectV2/Panel.cs

```diff
@@ -223,13 +223,13 @@ protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint,
             {
                 base.OnNewBeat(beatIndex, timingPoint, effectPoint, amplitudes);
 
-                if (beatIndex % separation != 0)
+                if ((beatIndex + 1) % separation != 0)
```

Comment:
I'm not sure what you think this is doing, but it's not doing what you think it's doing.